### ISSUE TITLE
Encounter end time is not set when closing the encounter

### ISSUE THREAD

**Issue by sathieshkumar** [drajer-health/eCRNow#506]
Encounter end time is not set while close the notification_context, set it to COMPLETED status. Currently end time is populated only at the launch api call.

Also in the **substituteContextParams** method, start time is used for end time.

```
      if (endDate == null) {
        if (data.getNotificationContext().getEncounterEndTime() != null) {
          endDate =
              DateFormatUtils.format(
                  DateUtils.addHours(
                      data.getNotificationContext().getEncounterStartTime(), endThreshold),
                  DATE_FORMAT);
        } else {
          endDate = DateFormatUtils.format(new Date(), DATE_FORMAT);
        }
      }
```

**Comment by nbashyam** [comment]
Already fixed in the baseline.

**Comment by VenkatSaiReddyTeegala** [comment]
Resolved as part of 3.1.4
